presto-record-decoder/src/main/java/io/prestosql/decoder/json/ISO8601JsonFieldDecoder.java

```diff
@@ -104,12 +111,12 @@ public long getLong()
 
             try {
                 String textValue = value.asText();
-                if (columnType.equals(TIMESTAMP)) {
+                if (columnType instanceof TimestampType) {
```

Comment:
Same comment as above. This needs to produce the proper encoding and type for the precision of TimestampType